src/web/web_views.nit

```diff
@@ -119,3 +119,41 @@ class HtmlDocPage
 		return tpl
 	end
 end
+
+# Display the source for each mentities
+class HtmlDotPage
+	super NitView
+
+	# Where to output dot images.
+	var out_file: String
+
+	# Dot to process.
+	var dot: Text
+
+	# Page title.
+	var title: String
+
+	redef fun render(srv) do
+		var tpl = new Template
+		tpl.add new Header(1, title)
+		tpl.add render_dot
+		return tpl
+	end
+
+	private fun render_dot: String do
+		var path = out_file
+		var file = new FileWriter.open("{path}.dot")
+		file.write(dot)
+		file.close
+		var proc = new ProcessReader("dot", "-Tsvg", "-Tcmapx", "{path}.dot")
+		var svg = new Buffer
+		var i = 0
+		while not proc.eof do
+			i += 1
+			if i < 6 then continue # skip dot default header
+			svg.append proc.read_line
+		end
+		proc.close
```

Comment:
the Process API is still crappy. you also need to call `wait` after the `close` (or you will get a zombie invasion)